app/controllers/surveys_controller.rb

```diff
@@ -80,4 +81,10 @@ def require_ownership
         redirect_to root_path, notice: "You are not the owner of that survey or an admin"
       end
     end
+
+    def require_published
+      path = @survey.published? ? survey_path(@survey) : root_path
+      redirect_to path, notice: 'Survey was successfully updated.'
```

Comment:
This is the wrong flash message, plus I think we can get this shorter, there's no need for both cases in the ternary statement

Reply:
Wow thanks @spencerldixon! Sorry that the message was wrong!